packages/react-components/src/text/src/Text.tsx

```diff
@@ -1,36 +1,39 @@
 import "./Text.css";
 
 import { Box } from "../../box";
-import { any, elementType, oneOf, oneOfType, string } from "prop-types";
-import { forwardRef } from "react";
-import { mergeClasses, mergeProps, normalizeSize, slot, useStyleProps } from "../../shared";
+import { ElementType, ForwardedRef, ReactElement, ReactNode, forwardRef } from "react";
+import { InnerPropsToProps, Size, mergeClasses, mergeProps, normalizeSize, slot, useStyleProps } from "../../shared";
 
-export function getTextClass(size) {
+export function getTextClass(size: Size): string {
     return `o-ui-text-${normalizeSize(size)}`;
 }
 
-////////
-
-const propTypes = {
+export interface InnerTextProps {
     /**
      * A text can vary in size.
      */
-    size: oneOf(["xs", "sm", "md", "lg", "xl", "2xl", "3xl", "4xl", "5xl", "inherit"]),
+    size?: "2xs" | "xs" | "sm" | "md" | "lg" | "xl" | "2xl" | "3xl" | "4xl" | "5xl" | "inherit";
     /**
      * An HTML element type or a custom React element type to render as.
      */
-    as: oneOfType([string, elementType]),
+    as?: ElementType;
     /**
-     * Default slot override.
+     * [Slot](?path=/docs/getting-started-slots--page) to render into.
      */
-    slot: string,
+    slot?: string;
     /**
-     * @ignore
+     * React children
```

Comment:
We only document children when it's required, otherwise it  shouldn't be in the docs.